ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/ListIndexTool.java

```diff
@@ -0,0 +1,577 @@
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.common.spi.tools.Parser;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@ToolAnnotation(ListIndexTool.TYPE)
+public class ListIndexTool implements Tool {
+    public static final String TYPE = "ListIndexTool";
+    // This needs to be changed once it's changed in opensearch core in RestIndicesListAction.
+    private static final int MAX_SUPPORTED_LIST_INDICES_PAGE_SIZE = 5000;
+    public static final int DEFAULT_PAGE_SIZE = 100;
+    private static final String DEFAULT_DESCRIPTION = String
+        .join(
+            " ",
+            "This tool gets index information from the OpenSearch cluster.",
+            "It takes 2 optional arguments named `index` which is a comma-delimited list of one or more indices to get information from (default is an empty list meaning all indices),",
+            "and `local` which means whether to return information from the local node only instead of the cluster manager node (default is false).",
+            "The tool returns the indices information, including `health`, `status`, `index`, `uuid`, `pri`, `rep`, `docs.count`, `docs.deleted`, `store.size`, `pri.store. size `, `pri.store.size`, `pri.store`."
+        );
+
+    @Setter
+    @Getter
+    private String name = ListIndexTool.TYPE;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+    @Getter
+    private String version;
+
+    private Client client;
+    @Setter
+    private Parser<?, ?> inputParser;
+    @Setter
+    private Parser<?, ?> outputParser;
+    @SuppressWarnings("unused")
+    private ClusterService clusterService;
+
+    public ListIndexTool(Client client, ClusterService clusterService) {
+        this.client = client;
+        this.clusterService = clusterService;
+
+        outputParser = new Parser<>() {
+            @Override
+            public Object parse(Object o) {
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        // TODO: This logic exactly matches the OpenSearch _cat/indices REST action. If code at
+        // o.o.rest/action/cat/RestIndicesAction.java changes those changes need to be reflected here
+        // https://github.com/opensearch-project/ml-commons/pull/1582#issuecomment-1796962876
```

Comment:
This comment still references _cat/indices, but as it's now _list/indices as a subclass there should be a bit more clarity about the relationship.

Reply:
Done